integration-tests-jvm/elasticsearch/src/test/java/org/apache/camel/quarkus/component/elasticsearch/it/ElasticsearchTest.java

```diff
@@ -0,0 +1,359 @@
+                .statusCode(201)
+                .extract()
+                .body()
+                .asString();
+
+        // Verify index exists
+        RestAssured.given()
+                .queryParam("component", component)
+                .queryParam("indexName", indexName)
+                .get("/elasticsearch/exists")
+                .then()
+                .statusCode(200)
+                .body(is("true"));
+
+        // Search data
+        Awaitility.await().pollInterval(50, TimeUnit.MILLISECONDS).atMost(10, TimeUnit.SECONDS).until(() -> {
+            String searchResult = RestAssured.given()
+                    .queryParam("component", component)
+                    .queryParam("indexKey", indexKey)
+                    .body("Super Fast")
+                    .get("/elasticsearch/search/json")
+                    .then()
+                    .statusCode(200)
+                    .extract()
+                    .body()
+                    .asString();
+            return searchResult.equals(indexValue);
+        });
+    }
+
+    @ParameterizedTest
+    @MethodSource("componentNames")
+    public void testElasticsearchMultiSearch(String component) {
+        String indexName = UUID.randomUUID().toString();
+        String indexKey = "camel-quarkus";
+        String indexValue = "Sub Atomic, Super Fast Camel Quarkus";
+
+        // Index data
+        RestAssured.given()
+                .queryParam("component", component)
+                .queryParam("indexName", indexName)
+                .queryParam("indexKey", indexKey)
+                .contentType(ContentType.TEXT)
+                .body(indexValue)
+                .post("/elasticsearch/index")
+                .then()
+                .statusCode(201)
+                .extract()
+                .body()
+                .asString();
+
+        // Verify index exists
+        RestAssured.given()
+                .queryParam("component", component)
+                .queryParam("indexName", indexName)
+                .get("/elasticsearch/exists")
+                .then()
+                .statusCode(200)
+                .body(is("true"));
+
+        // Search data
+        Awaitility.await().pollInterval(50, TimeUnit.MILLISECONDS).atMost(100, TimeUnit.SECONDS).until(() -> {
+            String hits = RestAssured.given()
+                    .queryParam("component", component)
+                    .queryParam("indexName", indexName)
+                    .queryParam("indexKey", indexKey)
+                    .body("Sub Atomic,Super Fast,Nonsense")
+                    .get("/elasticsearch/search/multi")
+                    .then()
+                    .statusCode(200)
+                    .extract()
+                    .body()
+                    .asString();
+            return hits.equals("2");
+        });
+    }
+
+    @SuppressWarnings("unused")
+    private static String[] componentNames() {
+        return new String[] { "elasticsearch" };
```

Comment:
In theory this is not required and we could just hard code the component name (using `to` instead of `toD`).

Or is it worth keeping in case we do eventually want to test the component with the Quarkus managed ES client?

Reply:
I left it there for the Quarkus managed ES client. But i agree it is not useful now at all. Maybe i should add comment there why it is done like that to that method ?

Reply:
I left comment, i hope it is OK.